src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -0,0 +1,53 @@
+struct Cone{T,TR,TP,TZ} <: AbstractVolumePrimitive{T}
+    rtop::TR
+    rbot::TR
+    φ::TP
+    z::TZ
+    
+    function Cone( ::Type{T},
+                   rtop::Union{T, <:AbstractInterval{T}},
+                   rbot::Union{T, <:AbstractInterval{T}},
+                   φ::Union{Nothing, <:AbstractInterval{T}},
+                   z::Union{T, <:AbstractInterval{T}}) where {T}
+        @assert typeof(rtop) == typeof(rbot) "r-Intervals in Cone are not type stable"
+        new{T,typeof(rtop),typeof(φ),typeof(z)}(rtop, rbot, φ, z)
+    end
+end
+
+#Constructors
+function Cone(;rtopMin = 0, rtopMax = 1, rbotMin = 0, rbotMax = 1, φMin = 0, φMax = 2π, zMin = -1/2, zMax = 1/2)
+    T = promote_type(typeof.((rtopMin, rtopMax, rbotMin, rbotMax, φMin, φMax, zMin, zMax))...)
+    rtop, rbot = if rtopMin == rbotMin == 0 
+        T(rtopMax), T(rbotMax)
+    else
+        T(rtopMin)..T(rtopMax), T(rbotMin)..T(rbotMax)
+    end
+    φ = mod(φMax - φMin, 2π) == 0 ? nothing : T(φMin)..T(φMax)
```

Comment:
This calculation of `φ` is also not good because `2π` is different for Float32 and Float64.

For instance, `mod(Float32(2π) - Float32(0), 2π) = 1.7484556025237907e-7` which is not zero.
Calling `Cone` with `φMin = Float32(0), φMax = Float32(2π)` will not result in `φ = nothing` like we would like it to.

We could write `mod(T(φMax) - T(φMin), T(2π))`
However, we do not force `T` to be a `Float`, so I guess it can also be an `Int`. This would give an `InexactError`.

Is it acceptable to force T to be a `Float`?

Reply:
`mod(T(φMax) - T(φMin), T(2π))` 
I did not specified `<: AbstractFloat` (we should replace SSD just with AbstractFloat) so far as for now we could in general also use than something with units, e.g. `CartesianPoint(2.3u"m"...)`. (Cylindrical coordinates make this a bit tricky though. But certainly doable.). 
But using `Int` does not make really any sense in general so we should not care about that.

Reply:
I will add `float` to `promote_type(typeof.((rMin, rMax, φMin, φMax, zMin, zMax))...)` to assure that `T` is a Float.
We might need to change this if we implement unit support, but right now, promoting the type of `r` and `φ` would make no sense with units (one is in `m`, the other in `rad`)

Reply:
This is, for now, fixed by adding `float(...)` before `promote_type`.
We might need to rework this for units in the future, but then also think of a reasonable way of promoting spatial and angular coordinates differently.